lib/app/features/notifications/push_notifications.dart

```diff
@@ -0,0 +1,168 @@
+import 'dart:io';
+
+import 'package:appwrite/models.dart' hide Log;
+import 'package:firebase_messaging/firebase_messaging.dart';
+import 'package:flutter/foundation.dart';
+import 'package:gordon_ferguson_app/app/config/logger.dart';
+import 'package:gordon_ferguson_app/app/config/router.dart';
+import 'package:gordon_ferguson_app/app/features/posts/data/wordpress_client.dart';
+import 'package:gordon_ferguson_app/app/features/posts/view/screens/post_detail_view.dart';
+import 'package:gordon_ferguson_app/app/features/settings/data/shared_preferences.dart';
+import 'package:riverpod_annotation/riverpod_annotation.dart';
+import 'package:shared_preferences/shared_preferences.dart';
+import 'package:appwrite/appwrite.dart';
```

Comment:
we moght want an abstraction layer rather than directly coupling to appwrite/ just an fyi